pkg/collector/corechecks/sbom/processor.go

```diff
@@ -210,49 +210,60 @@ func (p *processor) processContainerImagesRefresh(allImages []*workloadmeta.Cont
 	}
 }
 
-func (p *processor) processHostScanResult(result sbom.ScanResult) {
-	log.Debugf("processing host scanresult: %v", result)
+func (p *processor) processFilesystemScanResult(result sbom.ScanResult, kind, sbomID string, sbomType model.SBOMSourceType, errCallback func(err error) error) {
+	log.Debugf("processing %s scan result: %v", kind, result)
 	sbom := &model.SBOMEntity{
 		Status:             model.SBOMStatus_SUCCESS,
-		Type:               model.SBOMSourceType_HOST_FILE_SYSTEM,
-		Id:                 p.hostname,
+		Type:               sbomType,
+		Id:                 sbomID,
 		InUse:              true,
 		GeneratedAt:        timestamppb.New(result.CreatedAt),
 		GenerationDuration: convertDuration(result.Duration),
 	}
 
-	if result.Error != nil {
-		log.Errorf("Scan error: %v", result.Error)
-		sbom.Sbom = &model.SBOMEntity_Error{
-			Error: result.Error.Error(),
+	defer func() {
+		err := result.Error
+
+		if errCallback != nil {
+			if err = errCallback(err); err != nil {
+				log.Errorf("failed scan: %s", err)
+				return
```

Comment:
doesn't this return prevent the sbom to be put in the queue ?

Reply:
If the errCallback returns an error, yes.
This is to mimic the actual behaviour : prevent the SBOM to be put in the queue when we receive procfs.ErrNotFound